api/envoy/extensions/filters/http/oauth2/v3/oauth.proto

```diff
@@ -62,8 +62,17 @@ message OAuth2Credentials {
 
 // OAuth config
 //
-// [#next-free-field: 11]
+// [#next-free-field: 12]
 message OAuth2Config {
+  enum AuthType {
+    // The client_id and client_secret will be sent in the URL encoded request body.
```

Comment:
The other alternative here is making thie DEFAUlT = 0 and allowing that to indicate "client specific default behavior" from the server. That way we could could safely change the Envoy defualt via runtime later, without breaking the wire protoocl. Not sure if there is a strong case for this, so this is just a suggestion.

Reply:
@htuch, are you suggesting in the DEFAULT = 0 case we have another config where we have auth type defined for different auth providers. If that is the case we might need new "auth provider" field in the api do support it. Am I getting you right?

Reply:
No, just that we could add DEFAULT = 0 and let the implementatoin pick between 1 and 2.

Reply:
@htuch, Now I got you. even though auth server supports both 1 and 2 if a request includes auth information in header and encoded body then auth server is supposed to fail (as a validation step). In that way we should avoid sending auth credentials in both types under a default config.